addons/point_of_sale/models/pos_config.py

```diff
@@ -241,6 +241,8 @@ def _check_currencies(self):
             raise ValidationError(_("The invoice journal must be in the same currency as the Sales Journal or the company currency if that is not set."))
         if any(self.journal_ids.mapped(lambda journal: journal.currency_id and journal.currency_id != self.currency_id)):
             raise ValidationError(_("All payment methods must be in the same currency as the Sales Journal or the company currency if that is not set."))
+        if any(self.available_pricelist_ids.mapped(lambda pl: pl.company_id.id not in [False, self.company_id.id])):
+            raise ValidationError(_("All the pricelists must either belong to the company of the point of sale, or not having a company at all."))
```

Comment:
This won't trigger if you only change the company. You have to create a new method decorated with

```python
@api.constrains('company_id', 'available_pricelist_ids')
```